documentation/release_notes.rst

```diff
@@ -8,6 +8,91 @@ The Intel® oneAPI DPC++ Library (oneDPL) accompanies the Intel® oneAPI DPC++/C
+- Fixed an issue with ``histogram`` algorithm, reordering template parameters to match function parameter order. Users are encouraged to remove explicit specification of template parameters where they may exist from their histogram calls and instead add any explicit type conversions of their arguments as required.
+- ``radix_sort`` and ``radix_sort_by_key`` kernel templates residing in
+  the ``oneapi::dpl::experimental::kt::gpu::esimd`` now throw ``std::bad_alloc`` if they fail to allocate global memory.
+- Fixed a potential hanging occurring with ``radix_sort`` and ``radix_sort_by_key`` kernel templates residing in
+  the ``oneapi::dpl::experimental::kt::gpu::esimd`` namespace. 
+- Fixed documentation for ``sort_by_key`` algorithm, which used to be mistakenly described as stable, despite being unstable for host execution policies (``seq``, ``unseq``, ``par``, ``par_unseq``). It may become unstable for device execution policies in the future. If stability is required, use ``stable_sort_by_key`` instead. 
+- Resolved register spills that led to runtime crashes when calling ``sort`` with DPC++ execution policies on CUDA devices.
+- Allow passing iterators to C++20 random access ranges as algorithm arguments.
+- Fix issues caused by initialization of SYCL queues in predefined execution policies.
+
+Known Issues and Limitations
+----------------------------
+New in This Release
+^^^^^^^^^^^^^^^^^^^
+- When compiled with -O0 and executed on GPU devices ``histogram`` algorithms may provide incorrect results.
+- Inclusion of ``<oneapi/dpl/dynamic_selection>`` prior to ``<oneapi/dpl/random>`` may result in compilation errors. Include ``<oneapi/dpl/random>`` first as a workaround.
+- Incorrect results may occur when using oneapi::dpl::experimental::philox_engine algorithm with no predefined template parameters with `word_size` values other than 64 and 32.
+- Incorrect results or a synchronous SYCL exception may be observed with DPC++ execution policies when compiling the following algorithms with -O0 optimization: ``exclusive_scan``, ``inclusive_scan``, ``transform_exclusive_scan``, ``transform_inclusive_scan``, ``copy_if``, ``remove``, ``remove_copy``, ``remove_copy_if'``, ``remove_if``, ``partition``, ``partition_copy``, ``stable_partition``, ``unique``, ``unique_copy``, and ``sort``.
+- Calls to ``transform_inclusive_scan``, ``transform_exclusive_scan``, ``inclusive_scan``, and ``exclusive_scan`` using DPC++ execution policies with an initial element require that the value type of the provided input sequence is convertible to the type of the initial element.
+- The following algorithms when called with DPC++ execution policies may apply the user-provided predicate or equality operator for calls with no user predicate ~2N times: ``copy_if``, ``remove``, ``remove_copy``, ``remove_copy_if``, ``remove_if``, ``partition_copy``, ``unique``, and ``unique_copy``.
```

Comment:
Let's rephrase this line to @MikeDvorskiy's suggestion. I've changed the grammar a bit from the original suggestion and also added a second sentence to clarify that the predicate is applied `O(n)` times. I think stating the complexity of the algorithms is important. The standard uses `O(n)` for many of the other scan based algorithms, so I think we can align with that instead of saying ~2n.
```suggestion
- The complexity of the following algorithms with DPC++ execution policies may exceed the number of applications of the user-provided predicate or equality operator compared to the C++ standard requirements: ``copy_if``, ``remove``, ``remove_copy``, ``remove_copy_if``, ``remove_if``, ``partition_copy``, ``unique``, and ``unique_copy``. In all cases, the predicate or equality operator is applied ``O(n)`` times.
```

Reply:
Done.